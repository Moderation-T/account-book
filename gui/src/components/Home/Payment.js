import React from 'react';
import PropTypes from 'prop-types';

function Payment({ income, outcome }) {
  return (
    <div>
      <span className="income">收入：{income}</span>
      <span className="outcome">支出：{outcome}</span>
    </div>
  );
}

Payment.propTypes = {
  income: PropTypes.number,
  outcome: PropTypes.number,
};
export default Payment;
