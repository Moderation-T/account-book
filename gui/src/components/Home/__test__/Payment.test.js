import React from 'react';
import Payment from '../Payment';

import { shallow } from 'enzyme';

/** 测试用例说明
 * 1. 是否正确显示 outcome 与 income 的值
 */

const props = {
  outcome: 100,
  income: 10,
};

let wrapper;
describe('test Payment component', () => {
  //  每次运行单个测试用例的时候会执行这一步
  beforeEach(() => {
    wrapper = shallow(<Payment {...props}></Payment>);
  });

  it('should render the component to match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('component should render correct income&outcome number', () => {
    expect(wrapper.find('div .income').text()).toEqual(`收入：${props.income}`);
    expect(wrapper.find('div .outcome').text()).toEqual(`支出：${props.outcome}`);
  });
});
