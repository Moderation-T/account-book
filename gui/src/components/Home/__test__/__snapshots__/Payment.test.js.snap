// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Payment component should render the component to match snapshot 1`] = `ShallowWrapper {}`;
